go.mod

```
@@ -3,49 +3,50 @@ module github.com/kubeflow/training-operator
 go 1.22
 
 require (
-	github.com/go-logr/logr v1.4.1
+	github.com/go-logr/logr v1.4.2
 	github.com/google/go-cmp v0.6.0
-	github.com/onsi/ginkgo/v2 v2.17.1
-	github.com/onsi/gomega v1.32.0
+	github.com/onsi/ginkgo/v2 v2.19.0
+	github.com/onsi/gomega v1.33.1
 	github.com/open-policy-agent/cert-controller v0.10.1
 	github.com/prometheus/client_golang v1.18.0
-	github.com/sirupsen/logrus v1.9.0
+	github.com/sirupsen/logrus v1.9.3
 	github.com/stretchr/testify v1.9.0
 	go.uber.org/zap v1.27.0
-	k8s.io/api v0.29.3
-	k8s.io/apimachinery v0.29.3
-	k8s.io/client-go v0.29.3
-	k8s.io/code-generator v0.29.3
-	k8s.io/klog/v2 v2.110.1
+	k8s.io/api v0.29.5
+	k8s.io/apimachinery v0.29.5
+	k8s.io/client-go v0.29.5
+	k8s.io/code-generator v0.29.5
+	k8s.io/klog/v2 v2.120.1
 	k8s.io/kube-openapi v0.0.0-20231010175941-2dd684a91f00
-	k8s.io/utils v0.0.0-20230726121419-3b25d923346b
+	k8s.io/utils v0.0.0-20240502163921-fe8a2dddb1d0
 	sigs.k8s.io/controller-runtime v0.17.3
 	sigs.k8s.io/jobset v0.5.2
+	sigs.k8s.io/kueue v0.6.3
```

This is a pretty old version of Kueue, no?
```suggestion
	sigs.k8s.io/kueue v0.8.1
```
If we really need a older major release of Kueue, that is concerning.

---

No available to use the latest version due to existing v1 codes.

---

How do we update v1 to use a later version of Kueue? I am not even sure if 0.6 will be in support much longer.

---

Oh is it because we are still on 0.29 and bringing in Kueue will cause an update of the k8s libs?

Currently I didn't see any reference of Kueue in the go.mods.

---

> Oh is it because we are still on 0.29 and bringing in Kueue will cause an update of the k8s libs?
> 
> Currently I didn't see any reference of Kueue in the go.mods.

Yes, that's right. Currently, we use the v1.29 K8s client, but once we introduce the latest Kueue, the k8s client should be upgraded to the latest version.

---

As we discussed today's sync, we can not depend on the k/k code directly.
So, we just import the Kueue library. But, after https://github.com/kubernetes/kubernetes/pull/124609 is merged, we can get rid of Kueue library and depend on the native Kube library to calculate the container resource requests.

---

@tenzen-y Please create issue to track it.

---

Yes, sure.

---

Opened: https://github.com/kubeflow/training-operator/issues/2280